File: admin/src/modules/system/types/message.ts
export const MessageStatus = {
  Enabled: 1,
  Disabled: 2,
} as const

export type MessageStatus = (typeof MessageStatus)[keyof typeof MessageStatus]

export const MessageTemplateType = {
  Notification: 1,
  Todo: 2,
  Alert: 3,
} as const

export type MessageTemplateType =
  (typeof MessageTemplateType)[keyof typeof MessageTemplateType]

export const MessageReceiverType = {
  Role: 1,
  User: 2,
  Department: 3,
  Initiator: 4,
  Assignee: 5,
} as const

export type MessageReceiverType =
  (typeof MessageReceiverType)[keyof typeof MessageReceiverType]

export interface MessageTemplateItem {
  id: number
  code: string
  name: string
  title: string
  content: string
  type: MessageTemplateType
  variables: string
  sort: number
  status: MessageStatus
  is_system: boolean
  remark: string
  created_at: string
  updated_at: string
}

export interface MessageTemplateListQuery {
  page: number
  page_size: number
  keyword?: string
  type?: MessageTemplateType | 0
  status?: MessageStatus | 0
}

export interface MessageTemplateListResponse {
  items: MessageTemplateItem[]
  total: number
  page: number
  page_size: number
}

export interface CreateMessageTemplatePayload {
  code: string
  name: string
  title: string
  content: string
  type: MessageTemplateType
  variables: string
  sort: number
  status: MessageStatus
  is_system: boolean
  remark: string
}

export interface UpdateMessageTemplatePayload {
  name: string
  title: string
  content: string
  type: MessageTemplateType
  variables: string
  sort: number
  status: MessageStatus
  is_system: boolean
  remark: string
}

export interface UpdateMessageTemplateStatusPayload {
  status: MessageStatus
}

export interface MessageReminderItem {
  id: number
  code: string
  name: string
  trigger_event: string
  template_id: number
  template_code: string
  template_name: string
  channels: string
  receiver_type: MessageReceiverType
  receiver_values: string
  advance_minutes: number
  link_url: string
  sort: number
  status: MessageStatus
  is_system: boolean
  remark: string
  created_at: string
  updated_at: string
}

export interface MessageReminderListQuery {
  page: number
  page_size: number
  keyword?: string
  trigger_event?: string
  template_id?: number
  receiver_type?: MessageReceiverType | 0
  status?: MessageStatus | 0
}

export interface MessageReminderListResponse {
  items: MessageReminderItem[]
  total: number
  page: number
  page_size: number
}

export interface CreateMessageReminderPayload {
  code: string
  name: string
  trigger_event: string
  template_id: number
  channels: string
  receiver_type: MessageReceiverType
  receiver_values: string
  advance_minutes: number
  link_url: string
  sort: number
  status: MessageStatus
  is_system: boolean
  remark: string
}

export interface UpdateMessageReminderPayload {
  name: string
  trigger_event: string
  template_id: number
  channels: string
  receiver_type: MessageReceiverType
  receiver_values: string
  advance_minutes: number
  link_url: string
  sort: number
  status: MessageStatus
  is_system: boolean
  remark: string
}

export interface UpdateMessageReminderStatusPayload {
  status: MessageStatus
}

File: server/internal/platform/model/message.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// MessageConfigStatus 表示消息配置状态。
type MessageConfigStatus int

const (
	// MessageConfigStatusEnabled 表示配置已启用。
	MessageConfigStatusEnabled MessageConfigStatus = 1
	// MessageConfigStatusDisabled 表示配置已禁用。
	MessageConfigStatusDisabled MessageConfigStatus = 2
)

// MessageTemplateType 表示消息模板类型。
type MessageTemplateType int

const (
	// MessageTemplateTypeNotification 表示站内通知模板。
	MessageTemplateTypeNotification MessageTemplateType = 1
	// MessageTemplateTypeTodo 表示待办提醒模板。
	MessageTemplateTypeTodo MessageTemplateType = 2
	// MessageTemplateTypeAlert 表示告警提醒模板。
	MessageTemplateTypeAlert MessageTemplateType = 3
)

// MessageReceiverType 表示提醒接收人配置方式。
type MessageReceiverType int

const (
	// MessageReceiverTypeRole 表示按角色接收。
	MessageReceiverTypeRole MessageReceiverType = 1
	// MessageReceiverTypeUser 表示按指定用户接收。
	MessageReceiverTypeUser MessageReceiverType = 2
	// MessageReceiverTypeDepartment 表示按部门接收。
	MessageReceiverTypeDepartment MessageReceiverType = 3
	// MessageReceiverTypeInitiator 表示业务发起人接收。
	MessageReceiverTypeInitiator MessageReceiverType = 4
	// MessageReceiverTypeAssignee 表示业务负责人接收。
	MessageReceiverTypeAssignee MessageReceiverType = 5
)

// MessageTemplate 是消息模板表模型。
type MessageTemplate struct {
	ID        uint                `gorm:"primaryKey" json:"id"`
	Code      string              `gorm:"size:128;not null;uniqueIndex" json:"code"`
	Name      string              `gorm:"size:64;not null" json:"name"`
	Title     string              `gorm:"size:128;not null" json:"title"`
	Content   string              `gorm:"type:text;not null" json:"content"`
	Type      MessageTemplateType `gorm:"type:smallint;not null;default:1" json:"type"`
	Variables string              `gorm:"type:text;not null" json:"variables"`
	Sort      int                 `gorm:"not null;default:0" json:"sort"`
	Status    MessageConfigStatus `gorm:"type:smallint;not null;default:1" json:"status"`
	IsSystem  bool                `gorm:"not null;default:false" json:"is_system"`
	Remark    string              `gorm:"size:255;not null;default:''" json:"remark"`
	CreatedAt time.Time           `json:"created_at"`
	UpdatedAt time.Time           `json:"updated_at"`
	DeletedAt gorm.DeletedAt      `gorm:"index" json:"-"`
}

// TableName 固定消息模板表名。
func (MessageTemplate) TableName() string {
	return "sys_message_template"
}

// MessageReminder 是消息提醒规则表模型。
type MessageReminder struct {
	ID             uint                `gorm:"primaryKey" json:"id"`
	Code           string              `gorm:"size:128;not null;uniqueIndex" json:"code"`
	Name           string              `gorm:"size:64;not null" json:"name"`
	TriggerEvent   string              `gorm:"size:128;not null;index" json:"trigger_event"`
	TemplateID     uint                `gorm:"not null;index" json:"template_id"`
	Channels       string              `gorm:"size:128;not null;default:'notification'" json:"channels"`
	ReceiverType   MessageReceiverType `gorm:"type:smallint;not null;default:1" json:"receiver_type"`
	ReceiverValues string              `gorm:"type:text;not null" json:"receiver_values"`
	AdvanceMinutes int                 `gorm:"not null;default:0" json:"advance_minutes"`
	LinkURL        string              `gorm:"size:255;not null;default:''" json:"link_url"`
	Sort           int                 `gorm:"not null;default:0" json:"sort"`
	Status         MessageConfigStatus `gorm:"type:smallint;not null;default:1" json:"status"`
	IsSystem       bool                `gorm:"not null;default:false" json:"is_system"`
	Remark         string              `gorm:"size:255;not null;default:''" json:"remark"`
	CreatedAt      time.Time           `json:"created_at"`
	UpdatedAt      time.Time           `json:"updated_at"`
	DeletedAt      gorm.DeletedAt      `gorm:"index" json:"-"`
}

// TableName 固定消息提醒规则表名。
func (MessageReminder) TableName() string {
	return "sys_message_reminder"
}
